internal/worker/worker.go

```diff
@@ -248,7 +248,37 @@ func (w *BaseWorker) executeCommand(ctx context.Context, diceDBCmd *cmd.DiceDBCm
 			}
 			cmdList = append(cmdList, watchCmd)
 			isWatchNotification = true
+
+		case Unwatch:
+			// Generate the Cmd being unwatched. All we need to do is remove the .UNWATCH suffix from the command and pass
+			// it along as is.
+			// Modify the command name to remove the .UNWATCH suffix, this will allow us to generate a consistent
+			// fingerprint (which uses the command name without the suffix)
+			diceDBCmd.Cmd = diceDBCmd.Cmd[:len(diceDBCmd.Cmd)-8]
+			watchCmd := &cmd.DiceDBCmd{
+				Cmd:  diceDBCmd.Cmd,
+				Args: diceDBCmd.Args,
+			}
+			cmdList = append(cmdList, watchCmd)
+			isWatchNotification = false
+		}
+	}
+
+	// Unsubscribe Unwatch command type
+	if meta.CmdType == Unwatch {
+		command := cmdList[len(cmdList)-1]
+		w.cmdWatchSubscriptionChan <- watchmanager.WatchSubscription{
+			Subscribe:    false,
+			WatchCmd:     command,
+			AdhocReqChan: w.adhocReqChan,
+			Fingerprint:  command.GetFingerprint(),
```

Comment:
The UNWATCH command unsubscribes using the fingerprint and not the command. The usage is as follows:
```sh
> GET.WATCH k
1) GET
2) 12345 # Fingerprint
3) value
> GET.UNWATCH 12345
OK
```

Reply:
FIxed.